File: DemoMethods/Basic/Query2.cs
using System;
using System.Linq;
using System.Web.Http;
using DemoMethods.Entities;
using DemoMethods.Helpers;
using Raven.Client;
using Raven.Client.Linq;

namespace DemoMethods.Basic
{
    public partial class BasicController : ApiController
    {

        [HttpGet]
        [Demo("Query II")]
        public object Query2(string eid = "ALFKI")
        {
            using (var session = DocumentStoreHolder.Store.OpenSession())
            {
                RavenQueryStatistics stats;
                var q = 
                    from company in session.Query<Company>().Statistics(out stats)
                    where company.ExternalId == eid
                    select company;

                var result = q.FirstOrDefault();
                return new
                {
                    stats.IndexName,
                    Result = result
                };
            }
        }
    }
}

File: DemoServer/PythonDemo/demo_methods/Basic/Query2.py
from demo.holder import DocumentStoreHolder
from demo.entities import *


class Query2(object):
    @staticmethod
    def query2(externalId="ALFKI"):
        with DocumentStoreHolder.get_store().open_session() as session:
            result, stats = list(session.query(object_type=Company,
                                               with_statistics=True).
                                 where(ExternalId=externalId))

            return result[0] if result else None
